test/adaptive_load/step_controller_test.cc

```diff
@@ -0,0 +1,194 @@
+#include "envoy/registry/registry.h"
+
+#include "fake_plugins/fake_input_variable_setter/fake_input_variable_setter.h"
+#include "nighthawk/adaptive_load/input_variable_setter.h"
+
+#include "external/envoy/source/common/config/utility.h"
+
+#include "api/adaptive_load/benchmark_result.pb.h"
+#include "api/adaptive_load/input_variable_setter_impl.pb.h"
+#include "api/adaptive_load/step_controller_impl.pb.h"
+#include "api/client/options.pb.h"
+
+#include "adaptive_load/plugin_loader.h"
+#include "adaptive_load/step_controller_impl.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+namespace {
+
+nighthawk::adaptive_load::BenchmarkResult MakeBenchmarkResultOutsideThreshold() {
+  nighthawk::adaptive_load::BenchmarkResult result;
+  nighthawk::adaptive_load::MetricEvaluation* evaluation = result.add_metric_evaluations();
+  evaluation->set_threshold_score(-1.0);
+  evaluation->set_weight(10.0);
+  return result;
+}
+
+nighthawk::adaptive_load::BenchmarkResult MakeBenchmarkResultWithinThreshold() {
```

Comment:
It seems we can combine MakeBenchmarkResultWithinThreshold and MakeBenchmarkResultOutsideThreshold into a single MakeBenchmarkResult(bool within_threshold).

Reply:
Done, actually realized that the score could be the arg.